springfox-core/src/main/java/springfox/documentation/builders/OperationBuilder.java

```diff
@@ -232,9 +233,20 @@ public OperationBuilder tags(Set<String> tags) {
     return this;
   }
 
+  /**
+   * Sets the nickname for the operation
+   *
+   * @param nickname - a string for the nickname
+   * @return this
+   */
+  public OperationBuilder nickname(String nickname) {
+      this.nickname = defaultIfAbsent(nickname, this.nickname);
+      return this;
+  }
+
   public Operation build() {
     return new Operation(method, summary, notes, responseModel, uniqueId, position, tags, produces,
-        consumes, protocol, securityReferences, parameters, responseMessages, deprecated, isHidden);
+        consumes, protocol, securityReferences, parameters, responseMessages, deprecated, isHidden, nickname);
```

Comment:
@dilipkrish 

Currently, `uniqueId` is automatically generated based on HTTP verb and function name.

If I take the new `nickname` field out of the `OperationBuilder`, and instead use `uniqueId` as the `operationId` source in the `ServiceModelToSwagger2Mapper` class, which of the following should I do?
1. Do not generate a `uniqueId`, and populate it with `nickname` from `@ApiOperation`
2. Continue to generate a `uniqueId` as is but override it when a `nickname` is supplied in `@ApiOperation`
3. Other ?